python/src/aiconfig/Config.py

```diff
@@ -137,15 +137,22 @@ async def serialize(self, model_name: str, data: Dict,  prompt_name: str, params
         returns:
             Prompt | List[Prompt]: A prompt or list of prompts representing the input data
         """
+        event = CallbackEvent("on_serialize_start", __name__, {model_name: model_name, "data": data, "prompt_name": prompt_name, "params": params})
```

Comment:
Should think of a more robust way to get info in there aside from __name__ or __file__, e.g. using traceback lib. Not blocking